adapta/storage/blob/s3_storage_client.py

```diff
@@ -132,15 +148,53 @@ def download_blobs(
         filter_predicate: Optional[Callable[[...], bool]] = None,
     ) -> None:
         """
-        Not yet implemented in S3 Client
+        Downloads blobs from S3 storage to a local path.
+
+        :param blob_path: Path to blob
+        :param local_path: Local path to download the blobs to
+        :param threads: Number of threads to use for the download
+        :param filter_predicate: Optional callable to filter blobs
+        :return:
         """
-        raise NotImplementedError("Not yet implemented in S3StorageClient")
+        s3_path = cast_path(blob_path)
+        try:
```

Comment:
We try to avoid hiding exceptions from clients - you can wrap them, but you need to re-raise. Also in this case you are doing global try-except, I'd rather do a try on each blob or not do any exception handling at all and leave it to the client

Reply:
Got it, that is a good practice indeed. 

I've changed it now so it tries on each blob in [11247c6](https://github.com/SneaksAndData/adapta/pull/437/commits/11247c610a0a01c7dc2a7e504107bd99fcfda553), I hope it is ok this way